src/EditorFeatures/CSharpTest/Completion/CompletionProviders/OperatorCompletionProviderTests.cs

```diff
@@ -0,0 +1,664 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders
+{
+    public class OperatorCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        public OperatorCompletionProviderTests(CSharpTestWorkspaceFixture workspaceFixture) : base(workspaceFixture)
+        {
+        }
+
+        internal override Type GetCompletionProviderType()
+            => typeof(OperatorCompletionProvider);
+
+        // The suggestion is e.g. "+". If the user actually types "+" the completion list is closed. Operators therefore do not support partially written items.
+        protected override string? ItemPartiallyWritten(string? expectedItemOrNull) => "";
+
+        private static IEnumerable<string[]> BinaryArithmeticAndLogicalOperators()
+        {
+            yield return new[] { "+" };
+            yield return new[] { "&" };
+            yield return new[] { "|" };
+            yield return new[] { "/" };
+            yield return new[] { "^" };
+            yield return new[] { "%" };
+            yield return new[] { "*" };
+            yield return new[] { ">>" };
+            yield return new[] { "<<" };
+            yield return new[] { "-" };
+        }
```

Comment:
There are some operators missing in this list:

* We don't suggest [conditional logical operators](
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#conditional-logical-operators) `&&` and `||`
* We don't suggest [assignment operators](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#assignment-operators) `*=` `/=` `%=` `+=` `-=` `<<=` `>>=` `&=` `^=` `|=`
* We don't suggest [delegate combining operators](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#addition-operator) `+=` `-=`

Shall we do something about any of those?

Reply:
> We don't suggest conditional logical operators && and ||

We can always add later if needed.  Can keep out for now.

> We don't suggest assignment operators

Seems fine.  Users can get the operator help.  And, if this could be made into an assignment, we already have a feature that will prompt you for that.

> We don't suggest delegate combining operators += -=

Seems very niche.  We can leave out.

The important point of operators here is to both speed up writing `((Foo)someExpr)` and to help people with discovery of operators they *don't* normally know about.

Reply:
> The important point of operators here is to both speed up writing ((Foo)someExpr) and to help people with discovery of operators they don't normally know about.

Yes. That is the defining motivation of this feature.